### ISSUE TITLE
Declaratively define which collections a workflow belongs in

### ISSUE THREAD

**Issue by hexylena** [dockstore/dockstore#4472]
**Is your feature request related to a problem? Please describe.**
At the reminder of one of the outreach team, we're starting to think about pushing all of our workflows to Dockstore (~150 galaxy workflows), and the main blocker for us doing that is having to manually add them to our collections.

**Describe the solution you'd like**
I'd love to be able to declare which collections a workflow belongs in to help us scale things a bit better and ensure we don't accidentally forget to add workflows to their collections and give them the visibility we'd like. I recognise there's probably a data modelling issue from y'all's side where the relationship is the opposite, and collections point at workflows, rather than workflows pointing at collections, but, it'd still make my life easier for sure.

Maybe in a workflow that would look like:

```yaml
# Register workflows for Dockstore (https://dockstore.org/)
version: 1.2
workflows:
  - name: intro-short
    subclass: GALAXY
    primaryDescriptorPath: topics/introduction/tutorials/galaxy-intro-short/workflows/Galaxy-Workflow-galaxy-intro-short.ga
    collections:
      - gtn/trainingmaterial
      - gtn/introduction
  - name: gene-calling
    subclass: GALAXY
    primaryDescriptorPath: topics/genome-annotation/tutorials/gene-calling/workflows/main.ga
    collections:
      - gtn/trainingmaterial
      - gtn/genomeannotation
```

it feels 'weird' from the side of an organisation owner that I can write some yaml which results in a workflow created under my organisation, but I can't say "please place it in my own org's collections"

**Describe alternatives you've considered**
I've considered scripting this with the API, but the outreach member suggested filing this feature request, and if it's judged out of scope we'll probably end up writing something.

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/DOCK-1916)
┆Fix Versions: Open-ended research tasks
┆Issue Number: DOCK-1916
┆Sprint: Backlog
┆Issue Type: Story

**Comment by denis-yuen** [comment]
Thanks for the request, it is a good idea but there are a few issues on our side for sure. 

> it feels 'weird' from the side of an organisation owner that I can write some yaml which results in a workflow created under my organisation, but I can't say "please place it in my own org's collections"

To clarify, when registering a workflow on Dockstore, we keep track of what GitHub organization that workflow comes from and allow you to search for it using that information. However, the organization that has collections in Dockstore (sorry for the confusion) is separate which is why you can form collections which include workflows from github, bitbucket, gitlab, etc. at the same time. 
i.e. you may happen to have an organization that is named the same as a github or bitbucket organization, but in reality they're independent

Users always may have roles in that Dockstore org, some of which can update that organization and some of which cannot. https://docs.dockstore.org/en/stable/advanced-topics/organizations-and-collections.html?highlight=collections#organization-membership

In other words, I like the idea but it requires some thought on our end.
